### PR TITLE
First cut at auto-generating appendix

### PR DESCRIPTION
**Pull request by ekr**
The appendix kept getting out of sync, so autogenerate it

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -48,7 +48,11 @@ diff-commit: $(draft).txt $(draft)-$(COMMIT).txt
 $(next).md: $(draft).md
 	sed -e"s/$(basename $<)-latest/$(basename $@)/" $< > $@
 
-%.xml: %.md
+%.md2: %.md
+	- rm $@
```
Comment (martinthomson) [inline Makefile]: You shouldn't have to use rm here.  Try adding
.INTERMEDIATE: %.md2

And add *.md2 to your clean target.

File: mk-appendix.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                print_syntax(APPENDICES[m.group(2)])
+                del APPENDICES[m.group(2)]
+                print
+            else:
+                print l,
+    else:
+        print l,
+        m = re.match("\S", l)
+        if m is None:
+            CURRENT += l
```
Comment (martinthomson) [inline mk-appendix.py]: So a figure can't include a blank line if it is going to be included in the appendix?

**Comment by martinthomson** [comment]
Not how I would have done it, but if it works, then it's fine.

File: mk-appendix.py (hunk 2 above)
Comment (ekr) [inline mk-appendix.py]: No, it can. It can't include a line with a leading non-space character.

**Pull request merged**